### PR TITLE
Add BackOffice support for Catalogues

### PR DESCRIPTION
**Pull request by wujuu**
- dodac catalogi do elastic searcha (potrzebne? - nie mamy nigdzie searcha po katalogach)
- taggings - narazie zrobilem jako lista stringow zamiast dodawac kolejna relacje - moge pozmieniac jesli trzeba

### CODE REVIEW

File: app/services/catalogue/delete.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+class Catalogue::Delete
```
Comment (goreck888) [inline app/services/catalogue/delete.rb:3]: Add Inheritance from ApplicationService and then change Catalogue::Delete.new(...).call to Catalogue::Delete.call(...)

File: app/services/catalogue/pc_create_or_update.rb [hunk 2]
```diff
@@ -11,6 +11,7 @@ def initialize(eosc_registry_catalogue, status, modified_at)
     @error_message = "Catalogue haven't been updated. Message #{eosc_registry_catalogue}"
     @status = status
     @source_type = "eosc_registry"
+    @eid = eosc_registry_catalogue["id"]
```
Comment (goreck888) [inline app/services/catalogue/pc_create_or_update.rb:14]: Looks like unused variable

File: app/views/backoffice/catalogues/form/_basic.html.haml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    = f.input :legal_status,
+      collection: Vocabulary.where(type: "Vocabulary::LegalStatus"),
+      label_method: :name, value_method: :id,
+      input_html: { multiple: false, data: { choice: true } }
+    = f.input :hosting_legal_entity,
+               collection: Vocabulary.where(type: "Vocabulary::HostingLegalEntity"),
+               disabled: cant_edit_catalogue(:hosting_legal_entity),
+               label_method: :name,
+               value_method: :id,
+               input_html: { multiple: false, data: { choice: true } }
```
Comment (goreck888) [inline app/views/backoffice/catalogues/form/_basic.html.haml:37]: Add missing trailing line, probably that's why styles test fails

File: app/views/backoffice/catalogues/form/_classification.html.haml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  .card-body
+    = f.association :scientific_domains, disabled: cant_edit_catalogue([scientific_domain_ids: []]),
+              input_html: { data: { choice: true } },
+              collection: sds,
+              label_method: :first, value_method: :second
+    = f.input :tags, multiple: true, input_html: { class: "form-control" },
+              wrapper_html: { "data-form-target" => "tag_list" },
+              as: :array, disabled: cant_edit_catalogue([tags: []])
+    = link_to_add_array_field("catalogue", "tags")
+    -# = f.association :structure_type, disabled: cant_edit_catalogue([structure_type_ids: []]),
```
Comment (goreck888) [inline app/views/backoffice/catalogues/form/_classification.html.haml:27]: Why is it commented?

File: app/views/backoffice/catalogues/index.html.haml [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+                      class: "delete-icon float-right" do
+                  %i.far.fa-trash-alt
+
+                - if policy([:backoffice, catalogue]).edit?
+                  = link_to _("Edit"),
+                      edit_backoffice_catalogue_path(catalogue),
+                      class: "btn-sm btn-warning float-right mr-4"
+
+            %span.status{ class: catalogue.status }
+              = catalogue.status
```
Comment (goreck888) [inline app/views/backoffice/catalogues/index.html.haml:35]: Missing trailing newline

File: app/services/catalogue/pc_create_or_update.rb [hunk 6]
```diff
@@ -11,6 +11,7 @@ def initialize(eosc_registry_catalogue, status, modified_at)
     @error_message = "Catalogue haven't been updated. Message #{eosc_registry_catalogue}"
```
Comment (goreck888) [inline app/services/catalogue/pc_create_or_update.rb:11]: In Line 3 you can also add an inheritance from ApplicationService and use Catalogue::PcCreateOrUpdate.call in the Catalogue::PcCreateOrUpdateJob

**Review by goreck888 — COMMENTED** [review]
Looks good, there are some missing trailing lines and the Catalogue services should inherit from the ApplicationService, and there's a need to add breadcrumbs for consistent layout and functionality

**Comment by Marcelinna** [comment]
Add catalogues to list on the top
image

**Comment by Marcelinna** [comment]
I dont see field  Inclusion Criteria (*), Validation Process (*), End Of Life (*), Scope (*) 
image

**Comment by Marcelinna** [comment]
lack of Other on left side
image

**Comment by Marcelinna** [comment]
lack of Admin section 
image

**Comment by Marcelinna** [comment]
page doesn't work? https://marketplace-11.docker-fid.grid.cyf-kr.edu.pl/catalogues

**Comment by Marcelinna** [comment]
problem on the text on button "Add new affiliation"
image

**Comment by Marcelinna** [comment]
if the field is not filled, we do not show it
image

**Comment by Marcelinna** [comment]
When I import the catalogue from the PC - upstream should be set to eosc_registry not MP
image

**Comment by Marcelinna** [comment]
Despite adding Scientific domain in Provider component I don't see this field filled in MP
image
image

**Comment by Marcelinna** [comment]
Delete Structure Type, this filed doesnt exist 
image

**Comment by Marcelinna** [comment]
@wujuu we don't want to show the page for EOSC catalogue In Marketplace front office. Plase make for this catalogue page dont available for users

File: app/controllers/backoffice/catalogues_controller.rb [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+    if valid_model_and_urls? && @catalogue.save(validate: false)
+      redirect_to backoffice_catalogue_path(@catalogue), notice: "New catalogue created successfully"
+    else
+      render :new, status: :bad_request
+    end
+  end
+
+  def update
+    catalogue_duplicate = @catalogue.dup
```
Comment (goreck888) [inline app/controllers/backoffice/catalogues_controller.rb:48]: Why are we calling duplication of an object here?

**Comment by Marcelinna** [comment]
When I create Catalogue in BO
image

**Comment by Marcelinna** [comment]
-Structure Type should be deleted permanently
-When we don't have value, we don't show header (Affiliation and Scientific domain should not be visible)
image

**Review by Marcelinna — APPROVED** [review]

**Pull request merged**